docker/Dockerfile.leap.42.3

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright 2019, Intel Corporation
+#
+# 'recipe' for Docker to build an RPM
+#
+
+# Pull base image
+FROM opensuse/leap:42.3
+MAINTAINER daos-stack <daos@daos.groups.io>
+
+# use same UID as host and default value of 1000 if not specified
+ARG UID=1000
+
+# Add build user (to keep rpmbuild happy)
+ENV USER build
+RUN useradd -u $UID -ms /bin/bash $USER
+RUN groupadd -g $UID $USER
+
+# Install basic tools
+RUN zypper --non-interactive update
+
+# basic building
+ENV OPENSUSE_REPO="https://download.opensuse.org/repositories"
+ENV TOOLS_42_3A="openSUSE:/Tools/openSUSE_42.3/openSUSE:Tools.repo"
+ENV TOOLS_42_3="${OPENSUSE_REPO}/${TOOLS_42_3A}"
+ENV HPC_42_3A="science:/HPC/openSUSE_Leap_42.3/science:HPC.repo"
+ENV HPC_42_3="${OPENSUSE_REPO}/${HPC_42_3A}"
+
+RUN zypper --non-interactive ar -f ${TOOLS_42_3}; \
+    zypper --non-interactive ar -f ${HPC_42_3}; \
+    zypper --non-interactive --gpg-auto-import-keys refresh
+RUN zypper --non-interactive install autoconf automake build                  \
+                                     ca-certificates-mozilla curl createrepo  \
+                                     git libtool lsb-release make             \
+                                     perl-libwww-perl perl-LWP-Protocol-https \
+                                     perl-XML-Parser rpm-build sudo
+
+# need to run the build command as root, as it needs to chroot
+RUN echo "build ALL=(ALL) NOPASSWD: /usr/bin/build" > /etc/sudoers.d/build
+
+# Need the repo for munge
+ARG JENKINS_URL=""
+ENV JENKINS_PATH="${JENKINS_URL}job/daos-stack/job"
+ENV JENKINS_ART="lastSuccessfulBuild/artifact/artifacts"
+ARG MNGE_BRANCH="master"
+RUN zypper --non-interactive ar --gpgcheck-allow-unsigned -f \
+    ${JENKINS_PATH}/munge/job/${MNGE_BRANCH}/${JENKINS_ART}/leap42.3 munge; \
+    zypper --non-interactive ref munge
```

Comment:
Again, why at the container level and not just the `build` chroot level with a `--repo ...` argument?

Reply:
Trade off between putting distro specfiic stuff in a distro specific docker file or put it in the Jenkinsfile.
Or put a distro lookup table in the one of the Makefiles.

Looks easier to read/maintain currently to add the repositories to the docker container.

Reply:
But again, requires cachebusing and the churn of docker images associated with that to keep the repo data fresh.

Reply:
Added the zypper refresh command to the /etc/sudoers.d/build file and moved the refresh to the Makefile_packaging file.  Now no cachebusting is needed for the Suse Dockerfiles.

It also turns out that the for the science:HPC repository, if the repository URL without the .repo file is not passed as a --repo argument, build can not find the packages in it.

If the science:HPC.repo has not been added to the docker container repositories also, build can find the packages, but fails the rpm validation for those packages.